doc/decisions/manpages.md

```diff
@@ -11,17 +12,23 @@ Storing generated files is annoying, as it requires:
 
 ## Constraints
 
-1. we want beautiful rendered man pages
-2. we cannot require rare tools for the build process
+1. we want beautiful rendered manpages, e.g., OPTIONS section looks like normal manpages, see in Notes¹ below
+2. we cannot require rare tools for the build process: the manpages must be present in every package
 
 ## Assumptions
 
-- ronn-ng doesn't have packages on any distribution (violates constraint 2.)
-
 ## Considered Alternatives
 
-1. Doxygen: need to be rechecked in the latest version, see [#4551](https://issues.libelektra.org/4551)
-2. Pandoc 2.9 doesn't have the Markdown description-list feature like ronn-ng has (violates constraint 1.)
+0. `ronn-ng`, which doesn't have packages on most distribution (violates constraint 2.) and thus created this problem
+1. Write a tool that converts our specification, similar to [pythongen](/src/tools/pythongen/template/template.man)
+2. Write a tool that parses gopts `--help` output
+3. [help2man](https://www.gnu.org/software/help2man/)
+4. Doxygen:
+   - Constraint 1 probably broken
+5. Pandoc: has quite a few dependencies and would need rewrite of the current documentation in doc/help:
+   - Constraint 1 fulfilled via https://pandoc.org/MANUAL.html#definition-lists
+   - would need YAML metadata for every file
+     (it is possible to also pass this as command-line arguments via `--variable` but then we would move meta-information about manpages to the build system)
```

Comment:
First of all, the CMake code already needs to know which section a man page belongs to. We also use CMake to extract the proper date from the git log. So the build system is already somewhat involved.

To keep the information encoded in CMake scripts minimal, we could use a simple shell script that converts from

```markdown
# kdb-get(1) -- Get the value of a key stored in the key database
```

to

```markdown
---
title: KDB-GET
section: 1
...

## NAME

**kdb-get** - Get the value of a key stored in the key database
```

and pipe that output into `pandoc`. But I'd actually prefer just having the YAML front matter in the markdown file. Makes things more explicit and clear for beginners. It also get's rendered nicely as as table on GitHub. Random example I found: https://github.com/jonbeebe/frontmatter/blob/accd644b15a5ac7adaab261655eb552065020821/tests/testfile.md

Reply:
I did small clarifications in aec620f785ca38f1bdf0c954db9aaf59ed8a6c8a and added further possibilities to 5ee82b09cdd64a4192cb2daf6e128cc046aaae46

But as said: This PR was only to collect flying around information. Please add further information directly to the decision file after this was merged.